mmv1/products/vertexai/IndexEndpointDeployedIndex.yaml

```diff
@@ -0,0 +1,224 @@
+            - !ruby/object:Api::Type::String
+              name: projectId
+              output: true
+              description: |
+                Corresponding projectId in pscAutomationConfigs
+            - !ruby/object:Api::Type::String
+              name: network
+              output: true
+              description: |
+                Corresponding network in pscAutomationConfigs.
+            - !ruby/object:Api::Type::String
+              name: 'matchAddress'
+              output: true
+              description: |
+                ip Address created by the automated forwarding rule.
+  - !ruby/object:Api::Type::String
+    name: 'indexSyncTime'
+    output: true
+    description: If this timestamp's value is at least the Index.update_time of the original Index, it means that this DeployedIndex and the original Index are in sync. If this timestamp is older, then to see which updates this DeployedIndex already contains (and which it does not), one must list the operations that are running on the original Index.
+  - !ruby/object:Api::Type::NestedObject
+    name: automaticResources
+    description: A description of resources that the DeployedIndex uses, which to large degree are decided by Vertex AI, and optionally allows only a modest additional configuration.
+    ignore_read: true
+    properties:
+      - !ruby/object:Api::Type::Integer
+        name: 'minReplicaCount'
+        description: The minimum number of replicas this DeployedModel will be always deployed on.
+        default_value: 2
+      - !ruby/object:Api::Type::Integer
+        name: 'maxReplicaCount'
+        description: The maximum number of replicas this DeployedModel may be deployed on when the traffic against it increases.
+        custom_expand: templates/terraform/custom_expand/vertex_ai_endpoint_deployed_index.go.erb
+  - !ruby/object:Api::Type::NestedObject
+    name: dedicatedResources
+    description: A description of resources that are dedicated to the DeployedIndex, and that need a higher degree of manual configuration.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'machineSpec'
+        description: The minimum number of replicas this DeployedModel will be always deployed on.
+        immutable: true
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'machineType'
+            description: The type of the machine.
+            immutable: true
+      - !ruby/object:Api::Type::Integer
+        name: 'minReplicaCount'
+        description: The minimum number of machine replicas this DeployedModel will be always deployed on. This value must be greater than or equal to 1.
+        immutable: true
+        required: true
+      - !ruby/object:Api::Type::Integer
+        name: 'maxReplicaCount'
+        description: The maximum number of replicas this DeployedModel may be deployed on when the traffic against it increases.
+        immutable: true
```

Comment:
i've updated the basic test to include this. The only issue I'm currently looking into is when the user doesn't specify the use of `automatic_resources` since by default both min and max will be set to 2 if neither are set such as when their is no automatic_resources block.

For dedicated resources their is no default value set since it is a required field assuming the user is using the dedicated resources block. It's expected that the user must at least input the value for `minReplicaCount`. max will just default to min if nothing is set.